pkg/hub/csr/controller.go

```diff
@@ -105,18 +106,22 @@ func (c *csrApprovingController) sync(ctx context.Context, syncCtx factory.SyncC
 
 // Using SubjectAccessReview API to check whether a spoke agent has been authorized to renew its csr,
 // a spoke agent is authorized after its spoke cluster is accepted by hub cluster admin.
-func (c *csrApprovingController) authorize(ctx context.Context, csr *certificatesv1.CertificateSigningRequest) (bool, error) {
-	extra := make(map[string]authorizationv1.ExtraValue)
-	for k, v := range csr.Spec.Extra {
-		extra[k] = authorizationv1.ExtraValue(v)
+func (c *csrApprovingController) authorize(ctx context.Context, csr csrInfo) (bool, error) {
```

Comment:
@Promacanthus would also refactor this function to

```
// Using SubjectAccessReview API to check whether a spoke agent has been authorized to renew its csr,
// a spoke agent is authorized after its spoke cluster is accepted by hub cluster admin.
func authorize(ctx context.Context, kubeClient kubernetes.Interface, csr  *crsInfo) (bool, error) {
...
}
```

thus both v1 and v1beta1 can use it to authroize

Reply:
I will refactor the `authorize()` function.